tests/functional/defer_state/test_modified_state.py

```diff
@@ -261,3 +263,38 @@ def test_changed_exposure(self, project):
         results = run_dbt(["run", "--models", "+state:modified", "--state", "./state"])
         assert len(results) == 1
         assert results[0].node.name == "view_model"
+
+
+class TestChangedContract(BaseModifiedState):
+    def test_changed_contract(self, project):
+        self.run_and_save_state()
+
+        # update contract for table_model
+        write_file(contract_schema_yml, "models", "schema.yml")
+
+        # This will find the table_model node modified both through a config change
+        # and by a non-breaking change to contract: true
+        results = run_dbt(["run", "--models", "state:modified", "--state", "./state"])
+        assert len(results) == 1
+        assert results[0].node.name == "table_model"
+        manifest = get_manifest(project.project_root)
+        model_unique_id = "model.test.table_model"
+        model = manifest.nodes[model_unique_id]
+        expected_unrendered_config = {"contract": True, "materialized": "table"}
+        assert model.unrendered_config == expected_unrendered_config
+
+        # Run it again with "state:modified:contract", still finds modified due to contract: true
+        results = run_dbt(["run", "--models", "state:modified.contract", "--state", "./state"])
+        assert len(results) == 1
+        # save a new state
+        self.copy_state()
+
+        # This should raise because a column has added a constraint
```

Comment:
Adding a constraint shouldn't be a breaking change - just a regular change to the model contract (returns `False` from `same_contract`, but does not raise `ModelContractError`.